Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -474,28 +450,7 @@ std::unique_ptr<MaybeDocument> Serializer::DecodeBatchGetDocumentsResponse(
     // Ensure the tag matches the wire type
     switch (tag.field_number) {
       case google_firestore_v1beta1_BatchGetDocumentsResponse_found_tag:
-      case google_firestore_v1beta1_BatchGetDocumentsResponse_missing_tag:
-      case google_firestore_v1beta1_BatchGetDocumentsResponse_transaction_tag:
-      case google_firestore_v1beta1_BatchGetDocumentsResponse_read_time_tag:
-        if (tag.wire_type != PB_WT_STRING) {
-          reader->set_status(
-              Status(FirestoreErrorCode::DataLoss,
-                     "Input proto bytes cannot be parsed (mismatch between "
-                     "the wiretype and the field number (tag))"));
-        }
-        break;
-
-      default:
-        // Unknown tag. According to the proto spec, we need to ignore these. No
-        // action required here, though we'll need to skip the relevant bytes
-        // below.
-        break;
-    }
-
-    if (!reader->status().ok()) return nullptr;
-
-    switch (tag.field_number) {
-      case google_firestore_v1beta1_BatchGetDocumentsResponse_found_tag:
+        if (!reader->RequireWireType(PB_WT_STRING, tag)) return nullptr;
```

Comment:
In cases above, `NullValue` is returned. Is the difference deliberate?

Reply:
Yes; the cases above are in a function returning a (non pointer) FieldValue, whereas here we're returning a unique_ptr.

Reply:
Ah, sorry, I was looking at the diff and missed that.